### ISSUE TITLE
Using AjaxSelectMultipleField is very slow (on client-side?)

### ISSUE THREAD

**Issue by bepetersn** [pallets-eco/flask-admin#1259]
I've got a lot of related data that needs to be searched through for a field, `users`. I've configured an ajax model loader like so, pieced together from a few other issues:

```
user_loader = QueryAjaxModelLoader(
    'user', db.session, User, fields=['email', 'name'], page_size=10
)
form_extra_fields = {
    'users': AjaxSelectMultipleField(user_loader)
}
form_ajax_refs = {
    'user': user_loader
}
```

The page loads much faster, but I noticed the field itself doesn't behave very well client-side. Specifically, as soon as I click inside the select box (before I type anything), my whole browser freezes, for about 6 seconds. When it finishes loading, a dropdown is displayed, presumably with all possible options available. I don't see any calls to the /ajax/lookup/ API at this point, which is interesting. 

![location-page-users-field](https://cloud.githubusercontent.com/assets/1389463/14648624/69fe2208-0620-11e6-8fea-2acbd6831c83.png)

I'm looking for some direction for where (or whether) I might customize the underlying code so the AJAX calls don't start until after I start typing into the box, ideally after a minimum character limit. Is this functionality in the field, maybe? Or somewhere else.

**Comment by mrjoes** [comment]
That's very strange - by default Select2 is asynchronous and should not stall DOM updates.

Currently, there's no way to change how many characters are required - it is hard coded to minimum of 1 character and should not make any AJAX requests before typing single character: https://github.com/flask-admin/flask-admin/blob/master/flask_admin/static/admin/js/form.js#L14

**Comment by bepetersn** [comment]
Yeah, maybe I'm misinterpreting, because I assumed it wouldn't stall just on clicking. And there are no visible AJAX requests, also if makes any difference I was using the SQL_ECHO setting and saw no SQL firing while the stalling was happening.

**Comment by stamaimer** [comment]
I also meet this problem. I find [this](http://stackoverflow.com/questions/30713829/how-to-do-a-custom-query-in-model-view-on-to-many-relationship-in-flask-admin) in stackoverflow. 
`form_ajax_refs = {"user": {"fields": (User.email, User.name)}}` will work for you.

**Comment by ElLorans** [comment]
Closing this issue due to inactivity and lack of progress. If there's still interest or new context to add, feel free to reopen or open a new issue with updated details.
